addons/crm_livechat/models/discuss_channel.py

```diff
@@ -72,11 +72,13 @@ def _convert_visitor_to_lead(self, partner, key):
         })
 
     def _get_livechat_session_fields_to_store(self):
+        def can_read_opportunity_ids(self):
+            return self.env["crm.lead"].has_access("read")
         fields_to_store = super()._get_livechat_session_fields_to_store()
         fields_to_store.append(
             Store.Many(
                 "livechat_customer_partner_ids",
-                [Store.Many("opportunity_ids", ["id", "name"])],
+                [Store.Many("opportunity_ids", ["id", "name"], predicate=can_read_opportunity_ids)],
```

Comment:
This should not be done with a predicate when it doesn't depend on each record.
The field should simply not be added to the list in that case.